### PR TITLE
Add new SOwISC12to30E3r3 ocean and sea-ice mesh 

### PR DESCRIPTION
**Pull request by xylar**
Long name: SOwISC12to30kmL80E3SMv3r3

This version of the Southern Ocean Regionally Refined Mesh (SORRM) has resolution that is:
* 12 km around Antarctica
* 30 km elsewhere
It is intended to be similar to the IcoswISC30E3r5 mesh except in the Southern Ocean and around Antarctica.

This is a proposed E3SM v3 (E3) mesh for polar (formerly cryosphere) simulations.  This is revision 3 (r3) of the mesh, which includes many changes from revision 2 outlined in https://github.com/MPAS-Dev/compass/pull/807.  The mesh is tagged on Compass as: https://github.com/MPAS-Dev/compass/releases/tag/mesh_SOwISC12to30E3r3 to aid reproduction in the future.

The mesh and the G- and B-case results are being reviewed here:
https://acme-climate.atlassian.net/wiki/spaces/OO/pages/4441145345/Review+SOwISC12to30E3r3

A 30-year G- and 50-year B-case have been run, and analysis is posted on the review page.  Here are the simulation pages:
https://acme-climate.atlassian.net/wiki/spaces/PSC/pages/4575821825/20240829.GMPAS-JRA1p5-DIB-PISMF.TL319_SOwISC12to30E3r3.mesh-review-test.chrysalis
https://acme-climate.atlassian.net/wiki/spaces/PSC/pages/4575068161/20240904.v3.SORRME3r3.piControl.chrysalis.mesh-review

Relevant discussion can be found at https://github.com/E3SM-Ocean-Discussion/E3SM/pull/106

### CODE REVIEW

**Review by darincomeau — APPROVED** [review]
Approved based on test simulations referenced above and discussion in https://github.com/E3SM-Ocean-Discussion/E3SM/pull/106

**Comment by darincomeau** [comment]
I added a default `finidat` for 1850 conditions with the `r05` land grid, which was interpolated from the end of the v3 LR spinup.
10 year test using this finidat is here: https://acme-climate.atlassian.net/wiki/spaces/PSC/pages/4628250625/20240923.v3.SORRME3r3.piControl.lndIC.chrysalis
Notably there is no initial drastic drop in sea level as in the case with a blank `finidat`.

**Comment by xylar** [comment]
Rebased to fix conflicts with #6643

**Comment by xylar** [comment]
@alicebarthel, @cbegeman and @proteanplanet, could you please review and hopefully approve this week?

**Comment by cbegeman** [comment]
> I added a default finidat for 1850 conditions with the r05 land grid, which was interpolated from the end of the v3 LR spinup.
10 year test using this finidat is here: https://acme-climate.atlassian.net/wiki/spaces/PSC/pages/4628250625/20240923.v3.SORRME3r3.piControl.lndIC.chrysalis
Notably there is no initial drastic drop in sea level as in the case with a blank finidat.

I examined the land ice mass balance fields and they look as expected. We hope that the negative river runoff anomaly around Amery shown [here](https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.cbegeman/analysis/20240923.v3.SORRME3r3.piControl.lndIC.chrysalis/ocean/index.html#&gid=17&pid=1) will be reduced once we have a configuration with the deep firn model.

**Review by cbegeman — APPROVED** [review]
Approved on the basis of water column depth around ISC, examination of @proteanplanet's mesh figures, and review of simulation results in MPAS-Analysis pages linked above and separately generated surface mass flux figures in the SO.

**Review by alicebarthel — APPROVED** [review]
Approving after examining @proteanplanet's mesh figures, @darincomeau's and @cbegeman's MPAS-Analysis pages. 
The Ocean Heat uptake near Zapiola is visible immediately, but I could not find any mesh or topography irregularities in the region, so I don't think this mesh specifically enhances the dynamical problem.

**Review by proteanplanet — APPROVED** [review]
I have thoroughly reviewed this mesh and compared it with the IcoswISC30E3r5 mesh, including checking for integrity of the ice shelf component.   Differences include a minimum ocean depth of 12 m as compared to 20.1m in the IcoswISC30E3r5 mesh to accommodate greater refinement in ice shelf cavities. Figures are being updated at https://acme-climate.atlassian.net/wiki/spaces/OO/pages/edit-v2/4441145345 for later use in publications.

**Comment by proteanplanet** [comment]
An addendum to my acceptance of this PR. The final set of plots are now posted on the mesh review page at https://acme-climate.atlassian.net/wiki/spaces/OO/pages/4441145345/Review+SOwISC12to30E3r3.  Something I want to make sure is on everyone's minds is the nearly halving of the minimum depth in the Arctic for which we need to keep a close eye on how the sea ice evolves in SORRM runs:

MPAS_OceanIce_SOwISC12to30E3r3_IcoswISC30E3r5_zoom_bathymetry_56

The issue we need to consider in association with this mesh is the spreading functions applied for liquid and frozen runoff, which particularly affect the Arctic and are bathymetrically sensitive. I suggest we consider a similar set of functions to those applied to the E3SM V3 HR mesh, with which @cbegeman is familiar.  However, that's not a consideration in whether or not this mesh is ready for implementation.  It's now a matter of how we use this excellent mesh.   Thank you, @xylar, @darincomeau and @cbegeman for your hard work putting this together.

**Comment by darincomeau** [comment]
I pushed an adjustment to the Chrysalis 'M' PE layout to this: https://pace.ornl.gov/exp-details/199117

File: cime_config/config_grids.xml [hunk 1]
```diff
@@ -5402,10 +5496,15 @@
     </gridmap>
 
     <gridmap ocn_grid="RRSwISC6to18E3r5" rof_grid="r05">
-      <map name="ROF2OCN_ICE_RMAPNAME">cpl/cpl6/map_r05_to_RRSwISC6to18E3r5.cstmnn.r250e1250_58NS.20240328.nc</map>
+      <map name="ROF2OCN_ICE_RMAPNAME">cpl/cpl6/map_r05_to_RRSwISC6to18E3r5_cstmnn.r250e1250_58NS.20240328.nc</map>
```
Comment (jonbob) [inline cime_config/config_grids.xml:5499]: @xylar -- any idea why this line is showing up as changed? Blank space?

Comment (xylar) [reply cime_config/config_grids.xml:5499]: No, I don't believe it's blank space.  It looks like GitHub is glitching.

Comment (xylar) [reply cime_config/config_grids.xml:5499]: Ah, no, there's a period that got changed to an underscore.

**Comment by jonbob** [comment]
Passes:
* SMS_P960.T62_SOwISC12to30E3r3.GMPAS-IAF.chrysalis_intel
* ERP_Ld3.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel.allactive-pioroot1

and copied all new files to the inputdata repo. Merged to next

**Comment by xylar** [comment]
I also synced over (and chmodded) the files that belong in `inputdata/shared` and `diagnostics`.

**Comment by jonbob** [comment]
merged to master

**Comment by xylar** [comment]
Woot, woot!  Thanks @jonbob!

**Pull request merged**